consensus/safety-rules/src/safety_rules.rs

```diff
@@ -150,6 +150,28 @@ impl SafetyRules {
         updated
     }
 
+    pub(crate) fn observe_tc(
```

Comment:
We should update the `highest_timeout_round` when local timeout and send the timeout message, not receiving the TC.